docs/devguide/testing.asciidoc

```diff
@@ -1,29 +1,75 @@
+* https://github.com/elastic/beats-tester[Beats Tester]
+* https://github.com/elastic/e2e-testing[End-to-End Testing]
+
+==== Beats Tester
+Beats Tester uses Vagrant and Ansible to ensure Beats can be
+installed, run and assert some basic functionality like logging and
+metrics are working as expected.
+
+Currently it runs on long living branches and on `main` on a daily
+schedule. It does not block PRs, nor does it appear on PR checks.
+
+It is currently run by Jenkins on
+https://beats-ci.elastic.co/job/Beats/job/beats-tester/[Beats Tester
+orchestrator], failure notifications are sent to Slack on #ingest-notifications.
+
+===== Known issues
+* It can be flaky due to a design issue: on every run it will fetch
+some artifacts from the https://artifacts-api.elastic.co[artifacts API],
+however if another job has published a newer artifact for that
+branch/PR, then the job will fail with a checksum error
+* It uses an older version of Ansible
+* Very few people understand how it works
+
+==== End to End Testing
+End to End Testing is used by the Observability team and focus on
+Kubernetes tests. It should not be used by the Elastic-Agent
+team.
+
+=== Tests in the Beats repository
+There are three test suites:
+
+* Unit Tests written in Go
+* Integration tests written in Go
+* Tests written in Python (deprecated, replaced by integration tests in Go)
 
 The tests written in Go use the https://golang.org/pkg/testing/[Go Testing
-package]. The tests written in Python depend on https://docs.pytest.org/en/latest/[pytest] and require a compiled and executable binary from the Go code. The python test run a beat with a specific config and params and either check if the output is as expected or if the correct things show up in the logs.
+package]. The tests written in Python depend on
+https://docs.pytest.org/en/latest/[pytest] and require a compiled and
+executable binary from the Go code (the system test binary). The
+python test run a beat with a specific config and params and either
+check if the output is as expected or if the correct things show up in
+the logs.
 
 For both of the above test suites so called integration tests exists. Integration tests in Beats are tests which require an external system like Elasticsearch to test if the integration with this service works as expected. Beats provides in its testsuite docker containers and docker-compose files to start these environments but a developer can run the required services also locally.
 
-==== Running Go Tests
+==== Running Go Unit Tests
 
-The Go tests can be executed in each Go package by running `go test .`. This will execute all tests which don't don't require an external service to be running. To run all non integration tests for a beat run `mage unitTest`.
+The Go unit tests can be executed in each Go package by running `go test .`. This will execute all tests which don't don't require an external service to be running. To run all non integration tests for a beat run `mage unitTest`.
```

Comment:
```suggestion
The Go unit tests can be executed in each Go package by running `go test .`. This will execute all tests which don't require an external service to be running. To run all non integration tests for a beat run `mage unitTest`.
```